x-pack/plugins/apm/server/lib/helpers/convert_ui_filters/get_environment_ui_filter_es.ts

```diff
@@ -5,11 +5,14 @@
  */
 
 import { ESFilter } from '../../../../typings/elasticsearch';
-import { ENVIRONMENT_NOT_DEFINED } from '../../../../common/environment_filter_values';
+import {
+  ENVIRONMENT_NOT_DEFINED,
+  ENVIRONMENT_ALL,
+} from '../../../../common/environment_filter_values';
 import { SERVICE_ENVIRONMENT } from '../../../../common/elasticsearch_fieldnames';
 
 export function getEnvironmentUiFilterES(environment?: string): ESFilter[] {
-  if (!environment) {
+  if (!environment || environment === ENVIRONMENT_ALL.value) {
```

Comment:
What's the use case for having environment as an optional argument here?

Reply:
`getEnvironmentUiFilterES` is called from places where `environment` is optional. Those callers would have to implement this themselves if it was required:
```js
const envFilter = environment ? getEnvironmentUiFilterES(environment) : []
```

Reply:
(is my guess - haven't checked)

Reply:
I see. I guess what I'm arguing is that environment should never be undefined. It should always be ENVIRONMENT_ALL.value, ENVIRONMENT_NOT_DEFINED.value, or another non-empty string. But obviously that's a bigger and potentially painful change.

Reply:
Agree. I think right now the client will not send environment when all is selected. We should ofc always send a value regardless if "not defined", "all" or "<real env>" is selected.